sensorium/training/trainers.py

```diff
@@ -84,20 +129,26 @@ def full_objective(model, dataloader, data_key, *args, **kwargs):
         regularizers = int(
             not detach_core
         ) * model.core.regularizer() + model.readout.regularizer(data_key)
-        return (
-            loss_scale
-            * criterion(
-                model(args[0].to(device), data_key=data_key, **kwargs),
-                args[1].to(device),
-            )
-            + regularizers
+
+        tot_main_loss = loss_scale * criterion(
+            model(args[0].to(device), data_key=data_key, **kwargs),
+            args[1].to(device),
         )
+        return (tot_main_loss + regularizers), (tot_main_loss, regularizers)
 
     ##### Model training ####################################################################################################
+
+    """
```

Comment:
@Nisone2000  what is this? and why is it commented out?

Reply:
That's if we use the class DEC as they did it in the paper. They save the cluster centroids as a property of the model s.t. it gets automaticly updated in the optimising step. Right now I'm saving the cluster centroids in a tensor.